File: Chapter4/R/Example5.R
# Chapter 4 - Poisson Distribution
# Example 5: Comparing Poisson(λ) for λ = 2, 5, 10

# Parameters
lambdas <- c(2, 5, 10)

# Individual plots + comparison
par(mfrow = c(2, 2))

for (lambda in lambdas) {
  x   <- 0:(lambda * 3)
  pmf <- dpois(x, lambda)
  barplot(pmf, names.arg = x, col = 'darkgreen',
          xlab   = 'k', ylab = 'P(X = k)',
          main   = sprintf('Poisson(%d): Mean=%d, SD=%.2f',
                           lambda, lambda, sqrt(lambda)),
          border = 'black')
}

# Overlay comparison
x_max <- max(lambdas) * 3
plot(0:x_max, dpois(0:x_max, lambdas[1]),
     type = 'b', pch = 19, col = 'blue',
     xlab = 'k', ylab = 'P(X = k)',
     main = 'Distribution Comparison',
     ylim = c(0, 0.4), lwd = 2)
lines(0:x_max, dpois(0:x_max, lambdas[2]),
      type = 'b', pch = 19, col = 'red',   lwd = 2)
lines(0:x_max, dpois(0:x_max, lambdas[3]),
      type = 'b', pch = 19, col = 'green', lwd = 2)
legend('topright',
       legend = paste('lambda =', lambdas),
       col    = c('blue', 'red', 'green'), lwd = 2, pch = 19)

par(mfrow = c(1, 1))

File: Chapter4/Python/Example5.py
# Chapter 4 - Poisson Distribution
# Example 5: Comparing Poisson(λ) for λ = 2, 5, 10

import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import poisson

# Parameters
lambdas = [2, 5, 10]

# Individual plots + comparison
plt.figure(figsize=(12, 8))

for i, lam in enumerate(lambdas, 1):
    plt.subplot(2, 2, i)
    x   = np.arange(0, lam * 3)
    pmf = poisson.pmf(x, lam)
    plt.bar(x, pmf, color=f'C{i-1}', alpha=0.7, edgecolor='black')
    plt.xlabel('k')
    plt.ylabel('P(X = k)')
    plt.title(f'Poisson({lam}): Mean={lam}, SD={np.sqrt(lam):.2f}')
    plt.grid(axis='y', alpha=0.3)

# Overlay comparison
plt.subplot(2, 2, 4)
x_max = max(lambdas) * 3
for i, lam in enumerate(lambdas):
    x   = np.arange(0, x_max)
    pmf = poisson.pmf(x, lam)
    plt.plot(x, pmf, 'o-', label=f'lambda = {lam}', linewidth=2)

plt.xlabel('k')
plt.ylabel('P(X = k)')
plt.title('Distribution Comparison')
plt.legend()
plt.grid(alpha=0.3)
plt.tight_layout()
plt.show()
